src/components/ChatLayout/index.jsx

```diff
@@ -0,0 +1,346 @@
+    const [emailExpert, setEmailExpert] = useState('')
+    const user = parseJwt(readCookie(ACCESS_TOKEN))
+    const requestRef = collection(db, 'requests')
+    const expertRef = collection(db, 'expertOnline')
+    const [requestList, setRequestList] = useState([])
+    const [userStatus, setUserStatus] = useState([])
+    const [expertList, setExpertList] = useState([])
+    const handleChange = (email) => {
+        setEmailExpert(email)
+    }
+
+    useEffect(() => {
+        const queryMessage = query(
+            requestRef,
+            where('expertEmail', '==', user.Email)
+        )
+
+        const unsubscribe = onSnapshot(queryMessage, (snapshot) => {
+            let messages = []
+            snapshot.forEach((doc) => {
+                messages.push({ ...doc.data(), id: doc.id })
+            })
+            setRequestList(messages)
+        })
+        return () => unsubscribe()
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    useEffect(() => {
+        const queryMessage = query(expertRef, where('status', '==', 'isOnline'))
+        const unsubscribe = onSnapshot(queryMessage, (snapshot) => {
+            let messages = []
+            snapshot.forEach((doc) => {
+                messages.push({ ...doc.data(), id: doc.id })
+            })
+            setExpertList(messages)
+        })
+        return () => unsubscribe()
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    useEffect(() => {
+        const type = user.Role === 'user' ? 'userRequest.Email' : 'expertEmail'
+        const queryMessage = query(requestRef, where(type, '==', user.Email))
+
+        const unsubscribe = onSnapshot(queryMessage, (snapshot) => {
+            let messages = []
+            snapshot.forEach((doc) => {
+                messages.push({ ...doc.data(), id: doc.id })
+            })
+            setUserStatus(messages)
+        })
+        return () => unsubscribe()
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    useEffect(() => {
+        if (
+            userStatus.length !== 0 &&
+            user.Role === 'user' &&
+            userStatus[0].status === 'active'
+        ) {
+            setIsSelect(true)
+            setRoom(userStatus[0].room)
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [userStatus])
+    const addRequest = async () => {
+        if (!isRequest) {
+            await addDoc(requestRef, {
+                userRequest: user,
+                expertEmail: emailExpert,
+                status: 'pending',
+                room: '',
+            })
+        }
+    }
+
+    const removeRequest = async () => {
+        await deleteDoc(doc(requestRef, userStatus[0].id))
+        setIsRequest(false)
+        setEmailExpert('')
+    }
+
+    const updateRequest = async (id) => {
+        const randomRoom = Math.random().toString(36).substring(2, 8)

```

Comment:
I don't know what you mean in this case. What is `Math.random().toString(36).substring(2, 8)` used for?